rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java

```diff
@@ -213,14 +210,19 @@ private boolean allowByMinerNotBanned(Peer sender, Message message) {
     private boolean allowByMessageUniqueness(Peer sender, Message message) {
         Keccak256 encodedMessage = new Keccak256(HashUtil.keccak256(message.getEncoded()));
 
+        Set<Keccak256> receivedMessages = receivedPeerMessages.getOrDefault(sender, Collections.synchronizedSet(new HashSet<>()));
         boolean contains = receivedMessages.contains(encodedMessage);
 
         if (!contains) {
             if (message.getMessageType() == MessageType.BLOCK_MESSAGE || message.getMessageType() == MessageType.TRANSACTIONS) {
-                if (this.receivedMessages.size() >= MAX_NUMBER_OF_MESSAGES_CACHED) {
-                    this.receivedMessages.clear();
+                if (receivedMessages.size() >= MAX_NUMBER_OF_MESSAGES_CACHED) {
+                    receivedMessages.clear();
                 }
-                this.receivedMessages.add(encodedMessage);
+                if (this.receivedPeerMessages.isEmpty()) {
```

Comment:
If I'm getting it right using this condition only one sender would be added to the map.
I think we should check `this.receivedPeerMessages.containsKey(sender)` instead.

Reply:
Addressed